src/robot/libdocpkg/model.py

```diff
@@ -190,3 +228,155 @@ def _arg_to_dict(self, arg):
             'required': arg.required,
             'repr': unicode(arg)
         }
+
+
+class DataTypeCatalog(object):
+
+    def __init__(self):
+        self._enum_catalog = set()
+        self._typed_dict_catalog = set()
+
+    def __iter__(self):
+        for type_dooc in sorted(self._typed_dict_catalog | self._enum_catalog):
+            yield type_dooc
+
+    def __bool__(self):
+        return bool(self._enum_catalog or self._typed_dict_catalog)
+
+    __nonzero__ = __bool__
```

Comment:
Could also just implement `__nonzero__` and use `@py2to3` class decorator that you can import from `robot.utils`.